File: src/main/java/com/github/frcsty/frozenactions/actions/player/BungeeAction.java
package com.github.frcsty.frozenactions.actions.player;

import com.github.frcsty.frozenactions.actions.Action;
import com.github.frcsty.frozenactions.actions.ActionContext;
import com.google.common.io.ByteArrayDataOutput;
import com.google.common.io.ByteStreams;
import org.jetbrains.annotations.NotNull;

import java.util.Arrays;

public final class BungeeAction implements Action {

    @Override
    public @NotNull String getId() {
        return "BUNGEE";
    }

    @Override
    public void run(final @NotNull ActionContext context) {
        final ByteArrayDataOutput output = ByteStreams.newDataOutput();

        Arrays.asList("Connect", context.getData()).forEach(output::writeUTF);
        context.getPlayer().sendPluginMessage(context.getPlugin(), "BungeeCord", output.toByteArray());
    }
}

File: src/main/kotlin/com/github/frcsty/frozenactions/actions/player/BungeeAction.kt
@file:Suppress("UnstableApiUsage")

package com.github.frcsty.frozenactions.actions.player

import com.github.frcsty.frozenactions.actions.Action
import com.google.common.io.ByteStreams
import org.bukkit.entity.Player
import org.bukkit.plugin.Plugin
import java.util.*

object BungeeAction : Action {

    override val id = "BUNGEE"

    override fun run(plugin: Plugin, player: Player, data: String) {
        val output = ByteStreams.newDataOutput()
        Arrays.stream<String>(arrayOf("Connect", data)).forEach { s: String -> output.writeUTF(s) }

        player.sendPluginMessage(plugin, "BungeeCord", output.toByteArray())
    }
}